src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,336 @@
+                success = NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+        if (success) {
+            notificationTimestamp = DateTimeUtils.currentTimeMillis();
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification ...tripMonitorNotifications) {
+        for (TripMonitorNotification notification : tripMonitorNotifications) {
+            if (notification != null) notifications.add(notification);
+        }
+    }
+
+    /**
+     * Check whether monitored trip check should be skipped.
+     * TODO: Should this be a method on {@link MonitoredTrip}?
+     */
+    public static boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        ZoneId zoneId;
+        Optional<ZoneId> fromZoneId = getZoneIdForCoordinates(trip.from.lat, trip.from.lon);
+        if (fromZoneId.isEmpty()) {
+            String message = String.format(
+                "Could not find coordinate's (lat=%.6f, lon=%.6f) timezone for monitored trip %s",
+                trip.from.lat,
+                trip.from.lon,
+                trip.id
+            );
+            throw new RuntimeException(message);
+        } else {
+            zoneId = fromZoneId.get();
+        }
+        // Get current time and trip time (with the time offset to today) for comparison.
+        ZonedDateTime now = DateTimeUtils.nowAsZonedDateTime(zoneId);
+        // TODO: Determine whether we want to monitor trips during the length of the trip.
+        //  If we do this, we will want to use the itinerary end time (and destination zoneId) to determine this time
+        //  value. Also, we may want to use the start time leading up to the trip (for periodic monitoring) and once
+        //  the trip has started, switch to some other monitoring interval while the trip is in progress.
+        ZonedDateTime tripTime = ZonedDateTime.ofInstant(trip.itinerary.startTime.toInstant(), zoneId)
+            // Offset the trip time to today's date.
+            .withYear(now.getYear())
+            .withMonth(now.getMonthValue())
+            .withDayOfMonth(now.getDayOfMonth());
+        // TODO: Change log level.
+        LOG.info("Trip {} starts at {} (now={})", trip.id, tripTime.toString(), now.toString());
+        // TODO: This check may eventually make use of endTime and be used in conjunction with tripInProgress check (see
+        //  above).
+        boolean tripHasEnded = now.isAfter(tripTime);
+        // Skip check if trip is not active today.
+        if (!trip.isActiveOnDate(now)) {
+            // TODO: Change log level.
+            LOG.info("Trip {} not active today.", trip.id);
+            return true;
+        }
```

Comment:
This should probably be changed. What about trips that start at 12:01am? We should probably have some internal lead time from the beginning of the next possible trip time and count back a certain number of hours from that time and begin checking then instead.

Reply:
12:01 am to 03:00 am or so is considered previous day?